### ISSUE TITLE
State property will return a value of undefined if similarly named getter is configured

### ISSUE THREAD

**Issue by xorinzor** [posva/pinia#1698]
### Reproduction

Having a state of:

```
{
    testProperty: null
}
```

and having 2 getters:
```
testProperty: (state) => state.testProperty,
somethingElse: (state) => state.testProperty !== null
```

calling `storeObj.somethingElse()` will return `true` because `state.testProperty` will return `undefined`
`storeObj.testProperty()`  will also return `undefined`

### Steps to reproduce the bug

1. Define a store and create a property in the state
2. Add a getter with the same name as the property you just added in the state
3. Calling the getter will result in `undefined`
4. Changing the name of the getter with the identical name as the property in the state will make it return the correct value

### Expected behavior

According to [the docs](https://pinia.vuejs.org/core-concepts/getters.html) a getter should access the value of a state property when you use `state.<key>` and should get the value of another getter when you use `this.<getterName>` .

Having a identically named getter should not produce a value of `undefined` for the state property.
Alternatively, if it is undesired to have a similarly named state property and getter, Pinia should throw an error in the console.

### Actual behavior

using `state.<propertyName>` when a getter with an identical name as `propertyName` is set, will return a value of `undefined`.

**Comment by posva** [comment]
Duplicate of https://github.com/vuejs/pinia/issues/1331 and already implemented 😄

**Comment by xorinzor** [comment]
@posva just not released yet then? I'm on 2.0.22 and experiencing this.
